astropy/io/fits/column.py

```diff
@@ -2402,3 +2414,105 @@ def _convert_ascii_format(format, reverse=False):
             recformat += str(width)
 
         return recformat
+
+
+def _parse_tdisp_format(tdisp):
+    """
+    Parse the ``TDISPn`` keywords for ASCII and binary tables into a
+    ``(format, width, precision, exponential)`` tuple (the TDISP values
+    for ASCII and binary are identical except for 'Lw', which is only present in BINTABLE
+    extensions
+    """
+
+    # Use appropriate regex for format type
+    tdisp = tdisp.strip()
+    if tdisp[0] == 'E':
+        if tdisp[1] in ('N', 'S'):
+            tdisp_re = TDISP_RE_DICT['ENES']
+        else:
+            tdisp_re = TDISP_RE_DICT['EGD']
+    elif tdisp[0] in ('A', 'L'):
+        tdisp_re = TDISP_RE_DICT['A']
+    elif tdisp[0] in ('I', 'B', 'O', 'Z'):
+        tdisp_re = TDISP_RE_DICT['IBOZ']
+    elif tdisp[0] == 'F':
+        tdisp_re = TDISP_RE_DICT['F']
+    elif tdisp[0] in ('G', 'D'):
+        tdisp_re = TDISP_RE_DICT['EGD']
+    else:
+        raise VerifyError('Format {!r} is not recognized.'.format(tdisp))
+
+    match = tdisp_re.match(tdisp.strip())
+    if not match or match.group('formatc') is None:
+        raise VerifyError('Format {!r} is not recognized.'.format(tdisp))
+
+    formatc = match.group('formatc')
+    width = match.group('width')
+    precision = None
+    exponential = None
+
+    # Some formats have
+    if tdisp[0] in ('I', 'B', 'O', 'Z', 'F', 'E', 'G', 'D'):
+        precision = match.group('precision')
+        if precision is None:
+            precision = 1
+    if tdisp[0] in ('E', 'D', 'G') and tdisp[1] not in ('N', 'S'):
+        exponential = match.group('exponential')
+        if exponential is None:
+            exponential = 1
+
+    return formatc, width, precision, exponential
+
+
+def _fortran_to_python_format(format_type, width, precision):
+    """
+    Turn the TDISPn fortran format pieces into a final Python format string.
+    See the format_type definitions below. If codes is changed to take
+    advantage of the exponential specification, will need to add it as another
+    input parameter.
+
+    # mapping from TDISP format to python format (code)
+    # A: Character
+    # L: Logical (Boolean)
+    # I: 16-bit Integer
+    # B: Binary Integer
+    # O: Octal Integer
+    # Z: Hexadecimal Integer
+    # F: Float (64-bit; fixed decimal notation)
+    # EN: Float (engineering fortran format, exponential multiple of thee
+    # ES: Float (scientific, same as EN but non-zero leading digit
+    # E: Float, exponential notation
+    # D: Double-precision Floating Point
+    # G: Double-precision Floating Point
+    """
+
+    if format_type in ('A','L'):
+        return '{{:>{}}}'.format(width)
+    elif format_type == 'I':
+        # Can't predefine zero padding and space padding before hand without
+        # knowing the value being formatted, so grabbing precision and using
+        # that to zero pad, ignoring width
+        return '{{:0{}d}}'.format(precision)
```

Comment:
If we do place the formats in a dict as well, one could use named arguments, e.g.,
```
TDISP_FMT_DICT = {'I': '{{:0{precision}d}}'}
```
Then, it is possible to format them with all four possible numbers gotten out of the fortran format:
```
fmt = TDISP_FMT_DICT[format_type]
return fmt.format(precision=precision, width=width)
```
Here, it doesn't matter if you have extraneous keywords.